## Exercise 1

#### After scoring a touchdown, a football team may elect to attempt a two-point conversion, by running or passing the ball into the end zone. If successful, the team score two points. For instance football team, the probability that this play is successful is 0.40. Let X = 1 be a successful, X = 0 if not. Find the mean and variance of X.

## Exercise 2

#### True or False: If A and B are mutually exclusive
#### (a) P(A ∪ B) = 0
#### (b) P(A ∩ B) = 0
#### (c) P(A ∪ B) = P(A ∩ B)
#### (d) P(A ∪ B) = P(A) + P(B)

## Exercise 3

#### Let X ∼ Bin(5, 0.3). Find
#### (a) P(X ≥ 2)
#### (b) P(1 ≤ X ≤ 3)
#### (c) P(X = 0)
#### (d) E(X)
#### (e) V ar(X)

## Exercise 5

#### Consider the Curry 2016 2017.csv and Thompson 2016 2017.csv data-files. These two data files contains information related to Stephen Curry and Klay Thompson statistics from the 2016-2017 NBA season. In Python, answer the following:
#### (a) Using the pandas library, read both csv files into Python and create two data￾frames: curry and thompson, respectively.
#### (b) Using the results from Section 4.3, report the average points per game and their margin of errors of Curry and Thompson for 2016-2017 NBA season.
#### (c) Estimate the margin of error of the average points per game of Curry and Thompson via simulation. Use 1000 simulations. Are these results similar to the results from part (b)?
#### (d) Estimate the margin of error of the difference of average points per game of Curry and Thompson via simulation. Use 1000 simulations. Is the different in averages points per game statistical significant?

In [10]:
# (a)
import pandas as pd
import numpy as np

curry = pd.read_csv('Curry_2016_2017.csv')
thompson = pd.read_csv('Thompson_2016_2017.csv')

In [4]:
curry.columns

Index(['Rk', 'G', 'Date', 'Age', 'Tm', 'Unnamed: 5', 'Opp', 'Unnamed: 7', 'GS',
       'MP', 'FG', 'FGA', 'FG%', '3P', '3PA', '3P%', 'FT', 'FTA', 'FT%', 'ORB',
       'DRB', 'TRB', 'AST', 'STL', 'BLK', 'TOV', 'PF', 'PTS', 'GmSc', '+/-'],
      dtype='object')

In [3]:
thompson.head(1)

Rk  G        Date     Age   Tm Unnamed: 5  Opp Unnamed: 7  GS     MP  ...  \
0   1  1  2016-10-25  26-260  GSW        NaN  SAS    L (-29)   1  33:21  ...   

   DRB  TRB  AST  STL  BLK  TOV  PF  PTS  GmSc  +/-  
0    1    1    2    0    0    1   4   11   3.0   -6  

[1 rows x 30 columns]

In [6]:
# (b)
# Curry average points
curry['PTS'].mean()

25.303797468354432

In [11]:
# Curry margin of error
2*curry['PTS'].std()/np.sqrt(curry.shape[0])

1.7315276954705145

In [12]:
# Thompson average points
thompson['PTS'].mean()

22.333333333333332

In [14]:
# Thompson margin of error
2*thompson['PTS'].std()/np.sqrt(thompson.shape[0])

1.807103556972667

In [48]:
# (c)
# Estimating the margin error 
from sklearn.utils import resample

curry_mean_res = list()
thompson_mean_res = list()

for i in range(0,1000):
    
    # Simulating the data
    curry_sim = resample(curry['PTS'], replace = True)
    thompson_sim = resample(thompson['PTS'], replace = True)
    
    # Calculating the mean
    curry_mean = np.mean(curry_sim)
    thompson_mean = np.mean(thompson_sim)
    
    # Computing the mean
    curry_mean_res.append(curry_mean)
    thompson_mean_res.append(thompson_mean)

In [80]:
# Points average of simulated data for Curry 
np.mean(curry_mean_res)

25.2843417721519

In [81]:
# Calculating the Margin Error of the Curry's results
2*np.std(curry_mean_res, ddof = 1)

1.6948667635833066

25.28 +- 1.69 = (23.59, 26.97)

In [82]:
# Points average of simulated data of Thompson 
np.mean(thompson_mean_res)

22.2750641025641

In [83]:
# Calculating the Margin Error of the Thompson's results
2*np.std(thompson_mean_res, ddof = 1)

1.774522650391127

22.27 +- 1.77 = (20.5, 24.04)

#### The results seem to be similar from part (b)

In [45]:
# (d)
# Defining a list to store results
mean_diff = list()

for i in range(0,1000):
    
    #Simulating the data
    curry_sim = resample(curry['PTS'], replace = True)
    thompson_sim = resample(thompson['PTS'], replace = True)
    
    # Calculating the mean
    curry_mean = np.mean(curry_sim)
    thompson_mean = np.mean(thompson_sim)
    
    # Computing the difference in marging error of average points
    diff = curry_mean - thompson_mean
    mean_diff.append(diff)

In [88]:
# Computing average difference of average points
np.mean(ME_diff)

-0.08007394796318557

#### It is not statistical significant because in statistics, when the first number is zero, it is considered not statistically significant

In [89]:
# Computing the margin error of the simulated results
2*np.std(ME_diff, ddof = 1)

0.6140250331603221

In [ ]:
-0.08 +- 0.61 (-0.69, 0.53)